File: src/main/java/liqp/filters/Times.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Times extends Filter {

    /*
     * times(input, operand)
     *
     * multiplication
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        if(value == null) {
            value = 0L;
        }

        super.checkParams(params, 1);

        Object rhsObj = params[0];

        if (super.isInteger(value) && super.isInteger(rhsObj)) {
            return super.asNumber(value).longValue() * super.asNumber(rhsObj).longValue();
        }

        return super.asNumber(value).doubleValue() * super.asNumber(rhsObj).doubleValue();
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Times.kt
package liqp.filter

import liqp.context.LContext

class Times : LFilter() {

  /**
   * times(input, operand)
   *
   * multiplication
   */
  override fun onFilterAction(params: FilterParams, value: Any?, chain: FilterChainPointer, context: LContext): Any? {
    val num = context.asNumber(value)?.toDouble() ?: 0.0
    val rhsObj = context.asNumber(params[0])?.toDouble() ?: 0.0

    return num * rhsObj
  }
}
